internal/edgedbtypes/datetime_test.go

```diff
@@ -758,6 +759,32 @@ func TestUnmarshalDuration(t *testing.T) {
 	}
 }
 
+func TestNanosecondsDuration(t *testing.T) {
+	cases := []struct {
+		input    Duration
+		mustFail bool
+	}{
+		{Duration(math.MaxInt64), true},
+		{Duration(math.MaxInt64 / 100), true},
+		{Duration(math.MaxInt64 / 1000), false},
```

Comment:
Would you add a test case for the max and min `std::duration` values from EdgeDB as well as max+1 and min-1.

Reply:
I used min/max values from [tests of the Rust implementation](https://github.com/edgedb/edgedb-rust/blob/master/edgedb-protocol/tests/datetime_chrono.rs) after [this recommendation](https://www.edgedb.com/docs/reference/bindings/datetime#precision). Please let me know if I was mistaken :slightly_smiling_face:

Reply:
Sorry I got turned around. My intention was to have a test for the largest and smallest possible values for `AsNanoseconds()` and then one unit out side the valid range on both ends. The idea being that if someone changes this function in the future they won't accidentally shrink the window or change the error behavior.

So its not the min/max edgedb.Duration values as the input. It is the min/max time.Duration values as the output.

Reply:
Makes sense ! I added cases on both sides of each boundary, plus some cases for arbitrary values within range